src/mpyl/cli/artifacts.py

```diff
@@ -48,10 +52,28 @@ def pull(obj: CliContext, tag, pr):
 @click.option("--tag", "-t", help="Tag to build", type=click.STRING, required=False)
 @click.option("--pr", type=click.INT, help="PR number to fetch", required=False)
 @click.pass_obj
-def push(obj: CliContext, tag, pr):
+def push(obj: CliContext, tag: str, pr: int):
+    logger = logging.getLogger("mpyl")
     target_branch = tag if tag else f"PR-{pr}"
     if not target_branch:
         raise click.ClickException("Either --pr or --tag must be specified")
-    print("inside push: ", obj.config)
-    print("inside push: ", target_branch)
-    Artifacts(obj.repo).push(branch=target_branch)
+    artifact_repo_config: RepoConfig = RepoConfig.from_git_config(
+        obj.config["vcs"]["artifactRepository"]
+    )
+    artifact_repo_path = Path("../" + artifact_repo_config.folder)
+
+    try:
+        clone_repository(artifact_repo_config, artifact_repo_path)
+        artifact_repo = Repository(
+            config=artifact_repo_config, root_dir=artifact_repo_path
+        )
+        build_artifacts = BuildArtifacts(
+            logger=logger,
+            codebase_repo=obj.repo,
```

Comment:
Why does this need to be a repo? Isn't the local filesystem enough?

Reply:
I don't see how unless you're proposing a completely different solution? A lot of the logic in `BuildArtifacts` uses the repo and I think it looks pretty clean actually?